torchio/transforms/augmentation/intensity/random_simulation.py

```diff
@@ -0,0 +1,172 @@
+
+from typing import Union, Tuple, Optional, Dict, List
+import torch
+from ....torchio import DATA, TypeData, TypeRangeFloat, TypeNumber, AFFINE, INTENSITY, TYPE, LABEL
+from ....data.subject import Subject
+from ....data.image import Image
+from .. import RandomTransform
+
+MEAN_RANGE = (0.1, 0.9)
+STD_RANGE = (0.01, 0.1)
+
+
+class RandomSimulation(RandomTransform):
+    r"""Generate a random image from label maps.
+
+    Args:
+        label_keys: Keys designating the label images in the sample that
+            will be used to generate the new image.
+            If a single string is given, the corresponding label map is expected
+            to be a label encoding map.
+            Default will use all images labeled as label in the sample.
+        image_key: Key to which the new volume will be saved.
+            If this key corresponds to an already existing volume, zero elements from
+            the new volume will be filled with elements of the original volume.
+        coefficients: Dictionary containing the mean and standard deviation for
+            each label. For each value :math:`v`, if a tuple
+            :math:`(a, b)` is provided then
+            :math:`v \sim \mathcal{U}(a, b)`.
+            Default values are :math:`(0.1, 0.9)` for the means and
+            :math:`(0.01, 0.1)` for the standard deviations.
+        binary: Boolean to tell if label maps should be binarized.
```

Comment:
I would give the possibility to choose the threshold for binarization
binarize_threshold : default None (no binarization is performed), it defined the cutting threshold use to binarize the mask

Reply:
@fepegar 
> Maybe this can be expanded a bit. Something like "If True, values in the label maps greater than zero will be set to one".

This would typically aim to evaluate partial volume effect by comparing binary label maps and partial volume maps so changing all values greater than zero would probably be a bit too extreme. That's why a `0.5` threshold was used.

@romainVala 
> I would give the possibility to choose the threshold for binarization
binarize_threshold : default None (no binarization is performed), it defined the cutting threshold use to binarize the mask

I'm not sure it makes sense to have different threshold values but why not.

Reply:
Well, then you're binarizing something I would not call a label map, as they are not discrete values (i.e. labels).These are called [fractional labelmap](https://www.slicer.org/wiki/Documentation/Labs/Segmentations#Design_and_implementation) in 3D Slicer, [fuzzy volumes](http://www.bic.mni.mcgill.ca/ServicesAtlases/Colin27Highres) in MNI, [tissue probability maps](http://cslras.pbworks.com/f/Unified+Segmentation.pdf) in SPM, etc.

Reply:
I propose PV label map, this is closer to the generative model we have in mind by taking a weighted mean.

Thresholding is added to easily transform PV label in (binary) labelmap. And in this case we will have the same as lab2im
If your inputs are pure PV labels, then yes the best threshold is 0.5, (that should conserve the total volume)
So I agree I am not sure if on need to change the threshold (it may be the purpose of an other transform) but  including it, is easy to handle, so let it be ...